### PR TITLE
Modify usage of `sed` to be compatible with macOS

### PR DESCRIPTION
**Pull request by KingAkeem**
Mac OS uses a different `sed` command from GNU. In order to use the `-i` flag you must pass the extension along with it so I added the `.hpp` extension if darwin is detected as the OS.

### CODE REVIEW

**Comment by faheel** [comment]
Good find! But instead of using a modified command to make it work on a different OS, I think we should use those commands that work the same on multiple OSs.

So I suggest you change the `sed -i ...` line to the following:
```bash
sed "/#include \"*\"/d" "$release_file" > "$release_file.tmp"
mv "$release_file.tmp" "$release_file"
```

File: scripts/release.sh [hunk 1]
```diff
@@ -47,5 +47,5 @@ do
     printf "\n\n" >> "$release_file"
 done
 
-# delete includes for non-standard header files from the release file
```
Comment (faheel) [inline scripts/release.sh]: This comment is important :grimacing:

Comment (KingAkeem) [reply scripts/release.sh]: Sorry about that, adding it back now.

**Comment by faheel** [comment]
Thank you for contributing! :+1:

**Pull request merged**